File: golang/cmd/zone_records/zone_records.go
// Usage: TOKEN=token go run cmd/zone_records/zone_records.go example.com
package main

import (
	"context"
	"fmt"
	"os"
	"strconv"

	"github.com/dnsimple/dnsimple-go/v5/dnsimple"
	"golang.org/x/oauth2"
)

func main() {
	oauthToken := os.Getenv("TOKEN")
	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: oauthToken})
	tc := oauth2.NewClient(context.Background(), ts)

	client := dnsimple.NewClient(tc)
	client.BaseURL = "https://api.sandbox.dnsimple.com"

	// get the current authenticated account (if you don't know who you are)
	whoamiResponse, err := client.Identity.Whoami(context.Background())
	if err != nil {
		fmt.Printf("Whoami() returned error: %v\n", err)
		os.Exit(1)
	}

	accountId := strconv.FormatInt(whoamiResponse.Data.Account.ID, 10)

	listZoneRecordsResponse, err := client.Zones.ListRecords(context.Background(), accountId, os.Args[1], nil)
	if err != nil {
		fmt.Printf("ListRecords() returned error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("%+v\n", listZoneRecordsResponse)
}

File: csharp/Examples/ZoneRecords.cs
using dnsimple;

namespace DnsimpleExamples.Examples;

// Usage: TOKEN=your-token dotnet run -- zone_records example.com
public static class ZoneRecords
{
    public static void Run(string[] args)
    {
        var token = Environment.GetEnvironmentVariable("TOKEN");
        if (string.IsNullOrEmpty(token))
        {
            Console.Error.WriteLine("The TOKEN environment variable is required");
            Environment.Exit(1);
        }

        if (args.Length < 1)
        {
            Console.Error.WriteLine("A zone name argument is required");
            Environment.Exit(1);
        }

        var zoneId = args[0];

        var client = new Client();
        client.ChangeBaseUrlTo("https://api.sandbox.dnsimple.com");
        client.AddCredentials(new OAuth2Credentials(token));

        // get the current authenticated account (if you don't know who you are)
        var accountId = client.Identity.Whoami().Data.Account.Id;

        // list the records for the zone
        try
        {
            var records = client.Zones.ListZoneRecords(accountId, zoneId).Data;
            foreach (var record in records)
            {
                Console.WriteLine($"Record: Id={record.Id}, Name={record.Name}, Type={record.Type}, Content={record.Content}, Ttl={record.Ttl}");
            }
        }
        catch (DnsimpleException e)
        {
            Console.Error.WriteLine($"ListZoneRecords() returned error: {e.Message}");
            Environment.Exit(1);
        }
    }
}
